### PR TITLE
Correctly log circulation changes

### PR DESCRIPTION
**Pull request by leonardr**
This branch fixes a problem I noticed with the log messages triggered when circulation information changes. Because the message was triggered after the LicensePool's fields had been changed, it was always saying "[new value] => [new value]".

This branch also refactors the log message generation into its own method, and adds a unit test for it.

### CODE REVIEW

File: model.py [hunk 1]
```diff
@@ -5646,8 +5652,63 @@ def update_availability(
...
+            args.extend([edition.medium, 
+                         edition.title or "[NO TITLE]",
+                         edition.author or "[NO AUTHOR]",
+                         self.identifier]
+                    )
+        else:
+            message += '%s'
+            args.append(self.identifier)
+
+        def _part(message, string, old_value, new_value):
```
Comment (courte) [inline model.py]: You're not using message right now in this method, but you probably could be. You could also pass in `args` as well, and then get rid of lines 5693-4, 5698-9, 5703-4, and 5708-9. Either way works, though; it's up to you.

File: model.py [hunk 2]
```diff
@@ -5646,8 +5652,63 @@ def update_availability(
...
+    def circulation_changelog(self, old_licenses_owned, old_licenses_available,
+                              old_licenses_reserved, old_patrons_in_hold_queue):
+        """Generate a log message describing a change to the circulation.
+
+        :return: a 2-tuple (message, args) suitable for passing into 
+        logging.info or a similar method
+        """
+        edition = self.presentation_edition
+        message = 'CHANGED '
+        blah = object()
```
Comment (courte) [inline model.py]: You're not using this here.

**Comment by courte** [comment]
Couple small things, but otherwise it looks fly.

File: model.py (hunk 1 above)
Comment (leonardr) [inline model.py]: This is sort of how I originally wrote it, but I had scope problems with `message`. I've redone it as per your suggestion.

File: model.py (hunk 2 above)
Comment (leonardr) [inline model.py]: This is left over from my attempts to figure out the scope problem.

**Comment by courte** [comment]
:+1:

**Pull request merged**